app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt

```diff
@@ -421,16 +420,13 @@ class FahrplanFragment : Fragment(), SessionViewEventsHandler {
         actionBar.setListNavigationCallbacks(arrayAdapter, OnDaySelectedListener(numDays))
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        if (requestCode == FAHRPLAN_FRAGMENT_REQUEST_CODE && resultCode == AlarmTimePickerFragment.ALERT_TIME_PICKED_RESULT_CODE && data != null) {
-            val alarmTimesIndex = data.getIntExtra(AlarmTimePickerFragment.ALARM_PICKED_INTENT_KEY, 0)
-            onAlarmTimesIndexPicked(alarmTimesIndex)
-        }
-        super.onActivityResult(requestCode, resultCode, data)
-    }
-
     private fun showAlarmTimePicker() {
-        AlarmTimePickerFragment.show(this, FAHRPLAN_FRAGMENT_REQUEST_CODE)
+        AlarmTimePickerFragment.show(this, FAHRPLAN_FRAGMENT_REQUEST_KEY) { requestKey, result ->
+            if (requestKey == FAHRPLAN_FRAGMENT_REQUEST_KEY) {
+                val alarmTimesIndex = result.getInt(AlarmTimePickerFragment.ALARM_TIMES_INDEX_BUNDLE_KEY)
```

Comment:
Same here as above. If you do this, maybe you want a custom functions for both.

Reply:
:+1: Thx. Fixed in d868cad1337cbaec8270587c103ea44e360fbf22.